posthog/api/hog_function.py

```diff
@@ -163,29 +165,38 @@ def validate(self, attrs):
             attrs["filters"] = attrs.get("filters") or {}
             attrs["inputs_schema"] = attrs.get("inputs_schema") or []
             attrs["inputs"] = attrs.get("inputs") or {}
-
-        if "inputs_schema" in attrs:
-            attrs["inputs_schema"] = validate_inputs_schema(attrs["inputs_schema"])
-
-        if "inputs" in attrs:
-            inputs = attrs["inputs"] or {}
-            existing_encrypted_inputs = None
-
-            if instance and instance.encrypted_inputs:
-                existing_encrypted_inputs = instance.encrypted_inputs
-
-            attrs["inputs_schema"] = attrs.get("inputs_schema", instance.inputs_schema if instance else [])
-            attrs["inputs"] = validate_inputs(attrs["inputs_schema"], inputs, existing_encrypted_inputs, attrs["type"])
-
-        if "filters" in attrs:
-            if attrs["type"] in TYPES_WITH_COMPILED_FILTERS:
-                attrs["filters"] = compile_filters_bytecode(attrs["filters"], team)
-            elif attrs["type"] in TYPES_WITH_TRANSPILED_FILTERS:
-                compiler = JavaScriptCompiler()
-                code = compiler.visit(compile_filters_expr(attrs["filters"], team))
-                attrs["filters"]["transpiled"] = {"lang": "ts", "code": code, "stl": list(compiler.stl_functions)}
-                if "bytecode" in attrs["filters"]:
-                    del attrs["filters"]["bytecode"]
+            attrs["mappings"] = attrs.get("mappings") or []
+
+        # Used for both top level input validation, and mappings input validation
+        def validate_input_and_filters(attrs: dict, type: str):
+            if "inputs_schema" in attrs:
+                attrs["inputs_schema"] = validate_inputs_schema(attrs["inputs_schema"])
+
+            if "inputs" in attrs:
+                inputs = attrs["inputs"] or {}
+                existing_encrypted_inputs = None
+
+                if instance and instance.encrypted_inputs:
+                    existing_encrypted_inputs = instance.encrypted_inputs
+
+                attrs["inputs_schema"] = attrs.get("inputs_schema", instance.inputs_schema if instance else [])
+                attrs["inputs"] = validate_inputs(attrs["inputs_schema"], inputs, existing_encrypted_inputs, type)
+
+            if "filters" in attrs:
+                if type in TYPES_WITH_COMPILED_FILTERS:
+                    attrs["filters"] = compile_filters_bytecode(attrs["filters"], team)
+                elif type in TYPES_WITH_TRANSPILED_FILTERS:
+                    compiler = JavaScriptCompiler()
+                    code = compiler.visit(compile_filters_expr(attrs["filters"], team))
+                    attrs["filters"]["transpiled"] = {"lang": "ts", "code": code, "stl": list(compiler.stl_functions)}
+                    if "bytecode" in attrs["filters"]:
+                        del attrs["filters"]["bytecode"]
+
+        validate_input_and_filters(attrs, attrs["type"])
+
+        if "mappings" in attrs and attrs["mappings"] is not None:
```

Comment:
Would be good currently to explicitly raise if mappings are provided for non site-destinations - again thinking about keeping things clean just in case for the future follow up.